### ISSUE TITLE
[ARM*/Unix]  Cross build cmake config using platform variables

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#9635]
There are several places where the cmake config files use the build host PLATFORM arch instead of the TARGET arch.  

https://github.com/dotnet/coreclr/blob/master/src/ToolBox/SOS/lldbplugin/CMakeLists.txt#L18
https://github.com/dotnet/coreclr/blob/master/src/ToolBox/SOS/lldbplugin/CMakeLists.txt#L24
https://github.com/dotnet/coreclr/blob/master/src/ToolBox/SOS/lldbplugin/CMakeLists.txt#L28
https://github.com/dotnet/coreclr/blob/master/src/ToolBox/SOS/lldbplugin/CMakeLists.txt#L32

To workaround I have building native binaries on the target platform.

Either this should be cleaned up, or the cross build docs should be modified to to include skipnative and then build on target.

@rahku 
@janvorli

**Comment by janvorli** [comment]
@sdmaclea thank you for pointing it out. I have thought we have already fixed all the occurences. This needs to be fixed.

**Comment by janvorli** [comment]
Hmm, actually, I think I was mistaken. The TARGET has a different semantics than what you expect. The TARGET is used for building cross targetting tools. So e.g. crossgen that runs on x86, but generates ARM code. Here, the PLATFORM is x86 and TARGET is ARM.
When cross building e.g. for ARM64 on AMD64, the PLATFORM is CLR_PLATFORM_ARCH_ARM64, so it is not the host. And the lldb plugin uses a correct symbol.
In other words, when cross building, it is equivalent to building on the target platform from the CLR_PLATFORM_ARCH_XXX point of view.

**Comment by sdmaclea** [comment]
Hmmm...  I need to dig deeper to understand why CLR_PLATFORM_ARCH_ARM64 does not appear to be set when cross building....  Or perhaps it is not the only platform set.

**Comment by sdmaclea** [comment]
Added debug messages to confirm it is set correctly when cross compiling.  Closing.  Root cause of my issue was independent.  Will discuss in that PR.
